[quant][graphmode] Add new tensorlist observer for LSTM

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35916 [quant][graphmode] Add quantize_per_tensor.tensors
* #35894 [quant][graphmode] Insert Observers for dynamic LSTM
* **#35893 [quant][graphmode] Add new tensorlist observer for LSTM**

Summary:
LSTM operator inputs have tensor list for activations and weights.
In graph mode we need a new observer to work with tensor list

Test Plan:
python test/quantization/test_quantization.py ObserverTest
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20830389](https://our.internmc.facebook.com/intern/diff/D20830389)

supriyar committed:
[quant][graphmode] Add new tensorlist observer for LSTM

Summary:
LSTM operator inputs have tensor list for activations and weights.
In graph mode we need a new observer to work with tensor list

Test Plan:
python test/quantization/test_quantization.py ObserverTest
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/observer.py:
```diff
@@ -516,6 +516,49 @@ def calculate_qparams(self):
 
         return scale.to(dtype=torch.float), torch.tensor([nudged_zero_point])
 
+class _MinMaxTensorListObserver(MinMaxObserver):
```
do you need to do this for all observers?

jerryzh168 replied:
Could you add a comment here talking about the context? that we will want to refactor if we want to support other types of observer?

supriyar committed:
Update on "[quant][graphmode] Add new tensorlist observer for LSTM"


Summary:
LSTM operator inputs have tensor list for activations and weights.
In graph mode we need a new observer to work with tensor list

Test Plan:
python test/quantization/test_quantization.py ObserverTest
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (changes requested):
Please add a comment for the observerr

supriyar committed:
Update on "[quant][graphmode] Add new tensorlist observer for LSTM"


Summary:
LSTM operator inputs have tensor list for activations and weights.
In graph mode we need a new observer to work with tensor list

Test Plan:
python test/quantization/test_quantization.py ObserverTest
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
LG, thanks
